cli/node/daemon_test.go

```diff
@@ -121,6 +116,83 @@ func TestSocketDaemon_Listen(t *testing.T) {
 	require.Regexp(t, "^couldn't bind socket: listen unix /test.sock: bind:", err)
 }
 
+func TestSocketDaemon_ConnectivityTest_Listen(t *testing.T) {
+	dir, err := ioutil.TempDir(os.TempDir(), "dela-test-")
+	require.NoError(t, err)
+
+	defer os.RemoveAll(dir)
+
+	daemon := &socketDaemon{
+		socketpath:  filepath.Join(dir, "daemon.sock"),
+		actions:     &actionMap{},
+		closing:     make(chan struct{}),
+		readTimeout: 50 * time.Millisecond,
+	}
+
+	err = daemon.Listen()
+	require.NoError(t, err)
+
+	defer daemon.Close()
+
+	conn, err := net.DialTimeout("unix", daemon.socketpath, 1*time.Second)
+	require.NoError(t, err)
+	require.NoError(t, conn.Close())
+}
+
+func TestSocketDaemon_ConnClosedFromClient_Listen(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		// This test does not work on Windows.
```

Comment:
Consider adding more details to include if it's "by design" or not, and why.

Otherwise my natural tendency would be to remove this and analyze why as a sort of bug hunt.

Reply:
I refactored the test to use a fake connection instead. The issue is Windows having a different behaviour than Linux in terms of when/how it returns the errors. And this test is only there to test one log.